ansible_ai_connect/ai/api/model_pipelines/wca/pipelines_base.py

```diff
@@ -464,6 +467,17 @@ def post_request():
         return playbook, outline, warnings
 
 
+class WCABaseRoleGenerationPipeline(
+    WCABasePipeline, ModelPipelineRoleGeneration, metaclass=ABCMeta
+):
+
+    def __init__(self, inference_url):
+        super().__init__(inference_url=inference_url)
+
+    def invoke(self, params: RoleGenerationParameters) -> RoleGenerationResponse:
```

Comment:
I'd rather this was kept as:
```
def invoke(self, params: RoleGenerationParameters) -> RoleGenerationResponse:
    raise NotImplementedError
```
Then just override it in `pipelines_dummy.py`

At the moment you have the "dummy" implementation in the "base" class and override it for "SaaS" and "onprem" to not implemented (which seems [arse about face](https://en.wiktionary.org/wiki/arse_about_face)).